main/actions/src/main/scala/sbt/Tests.scala

```diff
@@ -266,8 +266,21 @@ object Tests {
     }
   def overall(results: Iterable[TestResult.Value]): TestResult.Value =
     (TestResult.Passed /: results) { (acc, result) => if (acc.id < result.id) result else acc }
-  def discover(frameworks: Seq[Framework], analysis: Analysis, log: Logger): (Seq[TestDefinition], Set[String]) =
-    discover(frameworks flatMap TestFramework.getFingerprints, allDefs(analysis), log)
+  def discover(frameworks: Seq[Framework], analysis: Analysis, log: Logger): (Seq[TestDefinition], Set[String]) = {
+    val (discoveredTestDefinitions, mains) = discover(frameworks flatMap TestFramework.getFingerprints, allDefs(analysis), log)
+    val enumeratedTestDefinitions = enumerate(frameworks)
+    (discoveredTestDefinitions ++ enumeratedTestDefinitions, mains)
+  }
+
+  def enumerate(frameworks: Seq[Framework]): Seq[TestDefinition] = {
+    val enumeratingFrameworks = frameworks.map(f => (f, scala.util.Try(f.getClass.getMethod("enumerateTests")))).filter(frameworkAndMethod => frameworkAndMethod._2.isSuccess)
```

Comment:
a second option here is to first update the test-interface JAr to have this method, and then invoke it here and catch the JVM link exception (so existing clients still work) rather than using reflection.